src/compiler/transformers/es6.ts

```diff
@@ -10,6 +10,133 @@ namespace ts {
         /** Enables substitutions for block-scoped bindings. */
         BlockScopedBindings = 1 << 1,
     }
+    /**
+     * If loop contains block scoped binding captured in some function then loop body is converted to a function.
+     * Lexical bindings declared in loop initializer will be passed into the loop body function as parameters,
+     * however if this binding is modified inside the body - this new value should be propagated back to the original binding.
+     * This is done by declaring new variable (out parameter holder) outside of the loop for every binding that is reassigned inside the body.
+     * On every iteration this variable is initialized with value of corresponding binding.
+     * At every point where control flow leaves the loop either explicitly (break/continue) or implicitly (at the end of loop body)
+     * we copy the value inside the loop to the out parameter holder.
+     *
+     * for (let x;;) {
+     *     let a = 1;
+     *     let b = () => a;
+     *     x++
+     *     if (...) break;
+     *     ...
+     * }
+     *
+     * will be converted to
+     *
+     * var out_x;
+     * var loop = function(x) {
+     *     var a = 1;
+     *     var b = function() { return a; }
+     *     x++;
+     *     if (...) return out_x = x, "break";
+     *     ...
+     *     out_x = x;
+     * }
+     * for (var x;;) {
+     *     out_x = x;
+     *     var state = loop(x);
+     *     x = out_x;
+     *     if (state === "break") break;
+     * }
+     *
+     * NOTE: values to out parameters are not copies if loop is abrupted with 'return' - in this case this will end the entire enclosing function
+     * so nobody can observe this new value.
+     */
+    interface LoopOutParameter {
+        originalName: Identifier;
+        outParamName: Identifier;
+    }
+
+    const enum CopyDirection {
+        ToOriginal,
+        ToOutParameter
+    }
+
+    const enum Jump {
+        Break       = 1 << 1,
+        Continue    = 1 << 2,
+        Return      = 1 << 3
+    }
+
+    interface ConvertedLoopState {
+        /*
+         * set of labels that occurred inside the converted loop
+         * used to determine if labeled jump can be emitted as is or it should be dispatched to calling code
+         */
+        labels?: Map<string>;
+        /*
+         * collection of labeled jumps that transfer control outside the converted loop.
+         * maps store association 'label -> labelMarker' where
+         * - label - value of label as it appear in code
+         * - label marker - return value that should be interpreted by calling code as 'jump to <label>'
+         */
+        labeledNonLocalBreaks?: Map<string>;
+        labeledNonLocalContinues?: Map<string>;
+
+        /*
+         * set of non-labeled jumps that transfer control outside the converted loop

```

Comment:
Each of these starts out explained as "set of", where really these are the types of jumps that this converted loop might need to deal with. So can you change each of these to "Types of non-labeled jumps..." etc.?